aws/resource_aws_cloudwatch_event_rule.go

```diff
@@ -168,26 +166,23 @@ func resourceAwsCloudWatchEventRuleRead(d *schema.ResourceData, meta interface{}
 	}
 	log.Printf("[DEBUG] Setting boolean state: %t", boolState)
 	d.Set("is_enabled", boolState)
-	if err := saveTagsCloudWatchEvents(conn, d, aws.StringValue(out.Arn)); err != nil {
+
+	tags, err := keyvaluetags.CloudwatcheventsListTags(conn, arn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for CloudWatch Event Rule (%s): %s", arn, err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().Map()); err != nil {
 		return fmt.Errorf("error setting tags: %s", err)
 	}
+
 	return nil
 }
 
 func resourceAwsCloudWatchEventRuleUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).cloudwatcheventsconn
 
-	if d.HasChange("is_enabled") && d.Get("is_enabled").(bool) {
```

Comment:
I was trying to understand how removing this code was not breaking `is_enabled` functionality 😅 and my guess is that when the original implementation was done years ago, the `PutRule` API `State` parameter was not properly disabling/enabling rules.

Reply:
I was scratching my head for a while after seeing this